File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:
        def valid_bst(root, left_bound, right_bound):
            if (root is None):
                return True
            elif (root.val <= left_bound) or (root.val >= right_bound):
                return False
            return valid_bst(root.left, left_bound, root.val) and valid_bst(root.right, root.val,right_bound)
        return valid_bst(root, float('-inf'), float('inf')) 

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public static boolean validCheck(TreeNode root, long leftbound, long rightbound)
    {
        
        if (root == null)
        {
            return true;
        }
        else if ((root.val >= rightbound) || (root.val <= leftbound))
        {
            return false;
        }
        return (validCheck(root.left, leftbound, root.val) && validCheck(root.right, root.val, rightbound));
    }
    public boolean isValidBST(TreeNode root) {
        long leftbound = Long.MIN_VALUE;
        long rightbound = Long.MAX_VALUE;
        return validCheck(root, leftbound, rightbound);
        
    }
}